services/teams-mcp/src/transcript/transcript.service.ts

```diff
@@ -0,0 +1,741 @@
+    const span = this.trace.getSpan();
+    span?.setAttribute('userProfileId', userProfileId.toString());
+
+    const payload = await SubscriptionRequestedEventDto.encodeAsync({
+      userProfileId,
+      type: 'unique.teams-mcp.transcript.lifecycle-notification.subscription-requested',
+    });
+
+    this.logger.debug(
+      { userProfileId: userProfileId.toString(), eventType: payload.type },
+      'enqueuing subscription requested event',
+    );
+
+    const published = await this.amqpconn.publish(MAIN_EXCHANGE.name, payload.type, payload, {});
+
+    span?.setAttribute('published', published);
+    span?.addEvent('event published to AMQP', {
+      exchangeName: MAIN_EXCHANGE.name,
+      eventType: payload.type,
+      published,
+    });
+
+    this.logger.log(
+      {
+        exchangeName: MAIN_EXCHANGE.name,
+        payload,
+        published,
+      },
+      `publishing "${payload.type}" event to AMQP exchange`,
+    );
+
+    if (!published) throw new Error(`Cannot publish AMQP event "${payload.type}"`);
+  }
+
+  @Span()
+  public async subscribe(userProfileId: TypeID<'user_profile'>): Promise<void> {
+    const span = this.trace.getSpan();
+    span?.setAttribute('userProfileId', userProfileId.toString());
+
+    this.logger.debug({ userProfileId: userProfileId.toString() }, 'starting subscription process');
+
+    const subscription = await this.db.query.subscriptions.findFirst({
+      where: and(
+        eq(subscriptions.internalType, 'transcript'),
+        eq(subscriptions.userProfileId, userProfileId.toString()),
+      ),
+    });
+
+    if (subscription) {
+      span?.addEvent('found managed subscription', { id: subscription.id });
+      this.logger.log({ id: subscription.id }, 'found managed subscription in DB');
+
+      const expiresAt = new Date(subscription.expiresAt);
+      const now = new Date();
+      const diffFromNow = expiresAt.getTime() - now.getTime();
+
+      span?.setAttribute('subscription.expiresAt', expiresAt.toISOString());
+      span?.setAttribute('subscription.diffFromNowMs', diffFromNow);
+
+      this.logger.debug(
+        { id: subscription.id, expiresAt, now: now, diffFromNow },
+        'managed subscription expiration',
+      );
+
+      const minimalTimeForLifecycleNotificationsInMinutes = 15;
+      if (diffFromNow < 0) {
+        span?.addEvent('subscription expired, deleting');
+
+        const result = await this.db
+          .delete(subscriptions)
+          .where(eq(subscriptions.id, subscription.id));
+
+        span?.addEvent('expired managed subscription yeeted', {
+          id: subscription.id,
+          count: result.rowCount ?? NaN,
+        });
+
+        this.logger.log(
+          { id: subscription.id, count: result.rowCount ?? NaN },
+          'expired managed subscription yeeted',
+        );
+      } else if (diffFromNow <= minimalTimeForLifecycleNotificationsInMinutes * 60 * 1000) {
```

Comment:
write a comment on how this usecase happened with the 15 minutes and the dequeue latency [..]